src/Utilities/TagWithDialog.test.js

```diff
@@ -28,30 +29,42 @@ describe('EntityTable', () => {
   describe('DOM', () => {
     it('should render with count', () => {
       const store = mockStore({});
-      const wrapper = mount(<TagWithDialog store={store} count={10} />);
+      const view = render(<TagWithDialog store={store} count={10} />);
+
       expect(
-        toJson(wrapper.find('TagWithDialog').first(), { mode: 'shallow' })
-      ).toMatchSnapshot();
+        screen.getByRole('button', {
+          name: '10',
+        })
+      ).toBeVisible();
+      expect(view.asFragment()).toMatchSnapshot();
     });
   });
 
   describe('API', () => {
-    it('should NOT call actions', () => {
+    it('should NOT call actions', async () => {
       const store = mockStore({});
-      const wrapper = mount(<TagWithDialog store={store} count={10} />);
-      wrapper.find('button').first().simulate('click');
+      render(<TagWithDialog store={store} count={10} />);
+
+      await userEvent.click(
+        screen.getByRole('button', {
+          name: '10',
+        })
+      );
       const actions = store.getActions();
       expect(actions.length).toBe(0);
     });
 
-    it('should call actions', () => {
+    it('should call actions', async () => {
       const store = mockStore({});
-      const wrapper = mount(
-        <TagWithDialog store={store} count={10} systemId="something" />
+      render(<TagWithDialog store={store} count={10} systemId="something" />);
+
+      await userEvent.click(
+        screen.getByRole('button', {
+          name: '10',
+        })
       );
-      wrapper.find('button').first().simulate('click');
       const actions = store.getActions();
-      expect(actions.length).toBe(2);
+      expect(actions.length).toBe(3);
```

Comment:
Just a note, nothing to do here.

These checks on the length of actions is odd. It probably will also fail each time a new test above is added that dispatches actions. In the future we should maybe replace this with something more deterministic.

Reply:
Yeah, good catch. I also used `shouldDispatch` together within this PR to assert on specific actions. I will replace it in the PF5/React 18 PR (based on this branch).